extensions/formats/geotools-vector/src/main/java/mil/nga/giat/geowave/format/geotools/vector/retyping/date/DateFieldRetypingSource.java

```diff
@@ -0,0 +1,99 @@
+package mil.nga.giat.geowave.format.geotools.vector.retyping.date;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import mil.nga.giat.geowave.format.geotools.vector.RetypingVectorDataPlugin.RetypingVectorDataSource;
+
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.geotools.feature.AttributeTypeBuilder;
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.feature.type.AttributeDescriptor;
+import org.opengis.feature.type.Name;
+
+public class DateFieldRetypingSource implements
+		RetypingVectorDataSource
+{
+	private final static Logger LOGGER = Logger.getLogger(DateFieldRetypingSource.class);
+
+	private final SimpleFeatureType typeIn;
+	private final Map<String, String> fieldNameToTimestampFormat;
+
+	public DateFieldRetypingSource(
+			final SimpleFeatureType typeIn,
+			final Map<String, String> fieldNameToTimestampFormat ) {
+		this.typeIn = typeIn;
+		this.fieldNameToTimestampFormat = fieldNameToTimestampFormat;
+	}
+
+	@Override
+	public Object retypeAttributeValue(
+			Object value,
+			Name attributeName ) {
+		Object outValue = value;
+		final String localName = attributeName.getLocalPart();
+		if (value != null && fieldNameToTimestampFormat.containsKey(localName)) {
+			try {
+				outValue = new SimpleDateFormat(
```

Comment:
Knowing the SampleDateFormat is not Thread safe, this approach is an easy answer.  Since ingest is a high-performance component, I would like to see a ThreadLocal cache of pre-built SimpleDateFormats.